src/layouts/slab.py

```diff
@@ -66,44 +69,125 @@ class NodeType(IntEnum):
 ORDER_BOOK_LAYOUT = cStruct(Padding(5), "account_flags" / ACCOUNT_FLAGS_LAYOUT, "slab_layout" / SLAB_LAYOUT, Padding(7))
 
 
+class SlabHeader(NamedTuple):
+    bump_index: int
+    free_list_length: int
+    free_list_root: int
+    root: int
+    leaf_count: int
+
+
+@dataclass
+class SlabNode:
+    pass
+
+
+@dataclass
+class SlabUninitializedNode(SlabNode):
+    pass
+
+
+@dataclass
+class SlabLeafNode(SlabNode):
+    owner_slot: int
+    fee_tier: int
+    key: int
+    owner: PublicKey
+    quantity: int
+    client_order_id: int
+
+
+@dataclass
+class SlabInnerNode(SlabNode):
+    prefix_len: int
+    key: int
+    children: List[int]
+
+
+@dataclass
+class SlabFreeNode(SlabNode):
+    next: int
+
+
+@dataclass
+class SlabLastFreeNode(SlabNode):
+    pass
```

Comment:
Let's remove this extra class and set `next` to some negative constant for `SlabFreeNode`.

Reply:
Wait...why do we need the free node or uninitialized node at all? Can they all just be `SlabNode`?

Reply:
When do we need to traverse the set of free slab nodes?

Reply:
Correct me if I'm wrong, but doing this seems like adding a lot more bloat..

Reply:
> Wait...why do we need the free node or uninitialized node at all? Can they all just be SlabNode?

Yes, they can. Doing that means we need to keep track of `next` and `isInitialized` in `SlabNode` which I think reduces clarity. Right now we basically have a one to one mapping from the `construct` return value to data class. Also, it's not much code anyway. `SlabUninitializedNode` and  `SlabLastFreeNode` are just two lines each.

> When do we need to traverse the set of free slab nodes?

I don't think we are using any of the `unintializedNode`, `freeNode` and `LastFreeNode`. But it also doesn't feel right to just disregard them.

> Correct me if I'm wrong, but doing this seems like adding a lot more bloat..
See above.

Reply:
I don't know. I think having 3 different classes that don't do anything is quite confusing. But if we *have* to have them, I like consensing `unintializedNode`, `freeNode` and `lastFreeNode` all as `SlabNode`, with a comment saying these three construct layouts all map to `SlabNode`.

Reply:
Also aren't we using unnecessary space storing all the `uninitialized` and `free` nodes in memory? And I think this makes the DFS slower too since we now have to traverse these extra nodes in `self._nodes`.